lib/ansible/galaxy/collection.py

```diff
@@ -0,0 +1,891 @@
+
+        shutil.copy(tar_filepath, tar_path)
+        collection_name = "%s.%s" % (collection_manifest['collection_info']['namespace'],
+                                     collection_manifest['collection_info']['name'])
+        display.display('Created collection for %s at %s' % (collection_name, to_text(tar_path)))
+
+
+def _get_mime_data(collection_path):
+    with open(collection_path, 'rb') as collection_tar:
+        data = collection_tar.read()
+
+    boundary = '--------------------------%s' % uuid.uuid4().hex
+    file_name = to_bytes(os.path.basename(collection_path), errors='surrogate_or_strict')
+    part_boundary = b"--" + to_bytes(boundary, errors='surrogate_or_strict')
+
+    form = [
+        part_boundary,
+        b"Content-Disposition: form-data; name=\"sha256\"",
+        to_bytes(secure_hash_s(data), errors='surrogate_or_strict'),
+        part_boundary,
+        b"Content-Disposition: file; name=\"file\"; filename=\"%s\"" % file_name,
+        b"Content-Type: application/octet-stream",
+        b"",
+        data,
+        b"%s--" % part_boundary,
+    ]
+
+    content_type = 'multipart/form-data; boundary=%s' % boundary
+
+    return b"\r\n".join(form), content_type
+
+
+def _wait_import(task_url, key, validate_certs):
+    headers = {}
+    if key:
+        headers['Authorization'] = "Token %s" % key
+
+    wait = 2
+    display.vvv('Waiting until galaxy import task %s has completed' % task_url)
+
+    while True:
+        resp = json.load(open_url(task_url, headers=headers, method='GET', validate_certs=validate_certs))
+
+        if resp.get('finished_at', None):
+            break
+
+        status = resp.get('status', 'waiting')
+        display.vvv('Galaxy import process has a status of %s, wait %d seconds before trying again' % (status, wait))
+        time.sleep(wait)
+
+    for message in resp.get('messages', []):
+        level = message['level']
+        if level == 'error':
+            display.error("Galaxy import error message: %s" % message['message'])
+        elif level == 'warning':
+            display.warning("Galaxy import warning message: %s" % message['message'])
+        else:
+            display.vvv("Galaxy import message: %s - %s" % (level, message['message']))
+
+    if resp['state'] == 'failed':
+        code = resp['error'].get('code', 'UNKNOWN')
+        description = resp['error'].get('description', "Unknown error, see %s for more details" % task_url)
+        raise AnsibleError("Galaxy import process failed: %s (Code: %s)" % (description, code))
+
+
+def _find_existing_collections(path):
+    collections = []
+
+    for namespace in os.listdir(path):
+        namespace_path = os.path.join(path, namespace)
+        if os.path.isfile(namespace_path):
+            continue
+
+        for collection in os.listdir(namespace_path):
+            collection_path = os.path.join(namespace_path, collection)
+            if os.path.isdir(collection_path):
```

Comment:
You can save some indentation by reverting this clause:
```suggestion
            if not os.path.isdir(collection_path):
            	continue
```